In [1]:
import pandas
import tensorflow as tf

PUTANJA = 'C:/Users/User/Desktop/chest-xray-dataset-test'# staviti na putanju foldera koji sadrzi chest_xray_test_dataset csv fajl
csv_fajl = pandas.read_csv(PUTANJA + r"\chest_xray_test_dataset.csv")

model = tf.keras.models.load_model('saved_model\\0-dense---32-nodes---1-conv---accuracy-95_08')# staviti na putanju foldera koji sadrzi model

KATEGORIJE = ['ZDRAV', 'VIRUS', 'BAKTERIJA']# menjanje ovog niza ce uticati na ispis odgovora (iako ga u ovom testu ne koristimo)
IMG_SIZE = 400# menjati samo ako pri promeni model koji to zahteva (svi modeli koji su okaceni na git zahtevaju IMG_SIZE 400)

In [2]:
import numpy as np
import cv2


train_y = np.ndarray(0)
ssi = 0
if True:# da ne bih morao da brisem tab ispred svake linije
    for index, row in csv_fajl.iterrows():
        if ssi < 11:
            if row['Label'] != 'Normal':
                if str(row['Label_2_Virus_category']) != 'nan':
                    if row['Label_2_Virus_category'] == 'Streptococcus':
                        #csv_fajl.at[str(index), 'Label'] = 3
                        train_y = np.append(train_y, 3)
                        #row['Label'] = 3
                    elif row['Label_2_Virus_category'] == 'COVID-19':
                        #csv_fajl.at[str(index), 'Label'] = 4
                        train_y = np.append(train_y, 4)
                        #row['Label'] = 4
                    elif row['Label_2_Virus_category'] == 'ARDS':
                        #csv_fajl.at[str(index), 'Label'] = 5
                        train_y = np.append(train_y, 5)
                        #row['Label'] = 5
                    elif row['Label_2_Virus_category'] == 'SARS':
                        #csv_fajl.at[str(index), 'Label'] = 6
                        train_y = np.append(train_y, 6)
                        #row['Label'] = 6
                    else:
                        pass
                    # print(row['Label'], " => " , row['Label_1_Virus_category'], "|", row['Label_2_Virus_category'])#, row['X_ray_image_name'])
                else:
                    if row['Label_1_Virus_category'] == 'Virus':
                        #csv_fajl.at[str(index), 'Label'] = 1
                        train_y = np.append(train_y, 1)
                        #row['Label'] = 1
                    elif row['Label_1_Virus_category'] == 'bacteria':
                        #csv_fajl.at[str(index), 'Label'] = 2
                        train_y = np.append(train_y, 2)
                        #row['Label'] = 2
                    else:
                        pass
                    # print(row['Label'], " => " , row['Label_1_Virus_category'], "|", row['Label_2_Virus_category'])#, row['X_ray_image_name'])
            else:
                #csv_fajl.at[str(index), 'Label'] = 0
                train_y = np.append(train_y, 0)
                #row['Label'] = 0

    print("Ucitao " + str(len(train_y)) + " objekata")

for epp in [10]:#[1, 2, 3, 5, 10, 20]:

    model = tf.keras.models.load_model('saved_model\\100ep' + str(epp))# staviti na putanju foldera koji sadrzi model



    
    
    csv_fajl.head()
    training_data = []
    ssi3 = 0
    ssi2 = 0;

    

    for index, row in csv_fajl.iterrows():
        img_name = row['X_ray_image_name']
        lokacija = PUTANJA + '/test/' + str(img_name)
        test_slika = cv2.imread(lokacija, cv2.IMREAD_GRAYSCALE)
        test_slika = cv2.resize(test_slika, dsize=(IMG_SIZE, IMG_SIZE), interpolation=cv2.INTER_CUBIC)
        test_slika = np.array(test_slika)
        test_slika = (np.expand_dims(test_slika, 0))
        test_slika = (np.expand_dims(test_slika,-1))

        prediction = model.predict(test_slika)

        if int(np.argmax(prediction[0])) != int(train_y[index]):
            ssi3 += 1
        else:
            ssi2 += 1


    print('Za epp ' + str(epp) + ', model je dobio' + str(ssi2) + ' tacnih odgovora, a ' + str(ssi3) + ' netacnih odgovora (' + str(round((ssi2)/(len(train_y))*100, 1)) + "% tacnosti)")


Ucitao 624 objekata
Za epp 10, model je dobio399 tacnih odgovora, a 225 netacnih odgovora (63.9% tacnosti)


Za epp 1, model je dobio342 tacnih odgovora, a 282 netacnih odgovora (54.8% tacnosti)
Za epp 2, model je dobio327 tacnih odgovora, a 297 netacnih odgovora (52.4% tacnosti)
Za epp 3, model je dobio398 tacnih odgovora, a 226 netacnih odgovora (63.8% tacnosti)
Za epp 5, model je dobio350 tacnih odgovora, a 274 netacnih odgovora (56.1% tacnosti)
Za epp 10, model je dobio399 tacnih odgovora, a 225 netacnih odgovora (63.9% tacnosti)
Za epp 20, model je dobio367 tacnih odgovora, a 257 netacnih odgovora (58.8% tacnosti)